### PR TITLE
Correct AFN node temperature compatible with zone air temperature

### LINKED ISSUE #10238: AFN node temperature is very different than zone air temperature
Issue overview
--------------
The defect file shows that the "crawlspace_unit1" AFN node temperature follows the "living_unit1" zone air temperature which is much different than its corresponding actual zone air temperature. A workaround created by @lgu1234 is to swap the order of the AFN:Zone object in the input file.

image

### Details
Some additional details for this issue (if relevant):
 - Platform: Windows
 - Version of EnergyPlus: tested in 9.6 and 23.1

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added: [defect_file.zip](https://github.com/NREL/EnergyPlus/files/12677946/defect_file.zip)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
 - Fixes #10238 
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

The defect file shows that the "crawlspace_unit1" AFN node temperature follows the "living_unit1" zone air temperature which is much different than its corresponding actual zone air temperature. A workaround created by @lgu1234 is to swap the order of the AFN:Zone object in the input file. 

The fix provided a correct AFN zone index. After fix, AFN node temperatures are compatible with zone air temperature.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ X] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [X ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp [hunk 1]
```diff
@@ -10302,7 +10302,11 @@ namespace AirflowNetwork {
                         ErrorsFound = true;
                     } else {
                         NodeFound(i) = true;
-                        AirflowNetworkNodeData(zoneNum).EPlusNodeNum = i;
+                        for (int iZone = 1; iZone <= AirflowNetworkNumOfZones; iZone++) {
+                            if (MultizoneZoneData(iZone).ZoneNum == zoneNum) {
+                                AirflowNetworkNodeData(iZone).EPlusNodeNum = i;
+                            }
```
Comment (lgu1234) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp:10308]: The zone number should be AFN zone index, instead of zone number from Zone object.

**Review by jasondegraw — APPROVED** [review]
@lgu1234 @Myoldmopar This fix looks correct to me. I wish the unit test was a bit shorter and didn't require so much in the way of static data, but I don't think there's a way around that.

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp (hunk 1 above)
Comment (Myoldmopar) [reply src/EnergyPlus/AirflowNetwork/src/Solver.cpp:10308]: OK I see it.  This accesses the right element of the AFN Node Data array.  Good.

**Comment by Myoldmopar** [comment]
This is a fine change, @lgu1234 .  Thanks for confirming @jasondegraw.  I pulled develop in and tested it locally with no problem.  Merging this.

**Pull request merged**